api/router/endpoint.go

```diff
@@ -0,0 +1,83 @@
+package router
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/go-chi/render"
+	"github.com/unweave/unweave/api/types"
+	"github.com/unweave/unweave/services/endpointsrv"
+)
+
+func NewEndpointRouter(endpoints endpointsrv.Service) *EndpointRouter {
+	return &EndpointRouter{
+		endpoints: endpoints,
+	}
+}
+
+type EndpointRouter struct {
+	endpoints endpointsrv.Service
+}
+
+func (e *EndpointRouter) EndpointRunCheckHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	endpointID := chi.URLParam(r, "endpointRef")
+
+	id, err := e.endpoints.RunEndpointEvals(ctx, endpointID)
+	if err != nil {
+		render.Render(w, r.WithContext(ctx), types.ErrHTTPError(err, "Failed to run endpoint evals"))
+		return
+	}
+
+	render.JSON(w, r, struct {
+		CheckID string `json:"check_id"`
+	}{CheckID: id})
+}
+
+func (e *EndpointRouter) EndpointCreate(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	projectID := chi.URLParam(r, "project")
+
+	var req types.EndpointCreate
+	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
+		render.Render(w, r, types.ErrHTTPBadRequest(err, "invalid request body"))
+		return
+	}
+
+	endpoint, err := e.endpoints.EndpointExecCreate(ctx, projectID, req.ExecID)
+	if err != nil {
+		render.Render(w, r, types.ErrInternalServer(err, "create endpoint failed"))
```

Comment:
You should probably use the [`types.ErrHTTPError`](https://github.com/unweave/unweave/blob/master/api/types/error.go#L99) here. It automatically handles returning the specific error code if it can be unwrapped or a 500 if not.  We might not even need to expose `types.ErrInternalServer`.